ixmp/utils.py

```diff
@@ -147,11 +146,16 @@ def pd_write(df, f, *args, **kwargs):
         writer.save()
 
 
-def numcols(df):
-    """Return the indices of the numeric columns of *df*."""
-    dtypes = df.dtypes
-    return [i for i in dtypes.index
-            if dtypes.loc[i].name.startswith(('float', 'int'))]
+def year_list(x):
+    """Return the elements of x that can be cast to year (int)."""
+    lst = []
+    for i in x:
+        try:
+            int(i)  # this is a year
```

Comment:
not sure but maybe better to check using regex (something like `^\d{4}$`) if we don't want negative or too small/large number

Reply:
in the MESSAGEix westeros tutorial, we use `years = [690, 700, 710, ...]`, so let's not be overly restrictive